rdfunit-junit/src/main/java/org/aksw/rdfunit/junit/RLOGStatement.java

```diff
@@ -0,0 +1,47 @@
+package org.aksw.rdfunit.junit;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import com.hp.hpl.jena.rdf.model.ResourceFactory;
+import org.aksw.rdfunit.tests.results.RLOGTestCaseResult;
+import org.aksw.rdfunit.tests.results.TestCaseResult;
+import org.junit.runners.model.Statement;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+class RLOGStatement extends Statement {
+
+    private final RdfUnitJunitStatusTestExecutor rdfUnitJunitStatusTestExecutor;
+    private final RdfUnitJunitTestCase testCase;
+
+    RLOGStatement(RdfUnitJunitStatusTestExecutor rdfUnitJunitStatusTestExecutor, RdfUnitJunitTestCase testCase) {
+        this.rdfUnitJunitStatusTestExecutor = rdfUnitJunitStatusTestExecutor;
+        this.testCase = testCase;
+    }
+
+    @Override
+    public void evaluate() throws Throwable {
+        final Collection<TestCaseResult> testCaseResults = rdfUnitJunitStatusTestExecutor.runTest(testCase);
+        final Collection<RLOGTestCaseResult> remainingResults = new ArrayList<>();
+        for (TestCaseResult t : testCaseResults) {
+            RLOGTestCaseResult r = (RLOGTestCaseResult) t;
+            if (!resourceIsPartOfInputModel(r)) {
+                continue;
+            }
+            remainingResults.add(r);
+        }
+        final StringBuilder b = new StringBuilder();
+        b.append(testCase.getTestCase().getResultMessage()).append(":\n");
+        for (RLOGTestCaseResult r : remainingResults) {
+            b.append("\t").append(r.getResource()).append("\n");
+        }
+        assertThat(b.toString(), remainingResults.isEmpty());
+    }
+
+    private boolean resourceIsPartOfInputModel(RLOGTestCaseResult r) {
+        return testCase.getTestInputModel().contains(
+                ResourceFactory.createResource(r.getResource()), null);
```

Comment:
This checks if the failing resource is part of the `@InputModel` to not report things coming from the Schema or Additional Data. This says: "does the input model contain a statement, where the subject is the failing resource?"

I wonder if we should change this to be: "does the input model contain a statement, which defines the failing resource?", i.e. something along:

```
inputModel.contains(failingResource, RDF.TYPE);
```